src/idl_parser.cpp

```diff
@@ -2596,6 +2600,20 @@ CheckedError Parser::ParseProtoFields(StructDef *struct_def, bool isextend,
         }
         EXPECT(']');
       }
+
+      if(type.enum_def && type.base_type != BASE_TYPE_UNION && field->value.constant == "0") {
+          if(!type.enum_def->FindByValue("0")) {
+              // Enum doesn't contain an implicit default, and a default was not
+              // provided. Set a default value.
+              const EnumVal* min_ev = type.enum_def->MinValue();
+              if(!min_ev) {
+                  return Error("No minimum value for enum \"" + type.enum_def->name + "\"");
+              }
+              field->value.constant = min_ev->name;
+          }
+      }
```

Comment:
Could you explain this code (conditions) and add a unit-tests for this code (min_ev==nullptr and min_e!=nullptr)?
Is this case possible with a correctly defined .proto?